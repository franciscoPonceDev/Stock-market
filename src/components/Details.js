import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { fetchDetailsFromAPI } from '../redux/details/details';
import DetailsItem from './DetailsItem';

const Details = () => {
  const dispatch = useDispatch();
  const detailsData = useSelector((state) => state.detailsReducer);
  const { idCompany } = useParams();

  useEffect(() => {
    dispatch(fetchDetailsFromAPI(idCompany));
  }, []);

  return (
    <section className="companies-container">
      {
        detailsData && detailsData.map(
          (company) => <DetailsItem key={company.name} company={company} />,
        )
      }
    </section>
  );
};

export default Details;
