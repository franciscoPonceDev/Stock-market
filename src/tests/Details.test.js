import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Details from '../components/Details';

describe('Testing Details', () => {
  it('Render the Details page, should match to the snapshot', () => {
    const list = renderer
      .create(
        <Provider store={store}>
          <Details />
        </Provider>,
      )
      .toJSON();
    expect(list).toMatchSnapshot();
  });
});
