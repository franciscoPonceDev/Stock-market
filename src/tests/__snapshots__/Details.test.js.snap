// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Details Render the Details page, should match to the snapshot 1`] = `
<section
  className="companies-container"
/>
`;
